cli/daemon/run/proc_groups.go

```diff
@@ -71,6 +73,18 @@ func (pg *ProcGroup) Done() <-chan struct{} {
 	return c
 }
 
+// Start starts all the processes in the group.
+func (pg *ProcGroup) Start() (err error) {
+	for _, p := range pg.allProcesses {
```

Comment:
There's a deadlock I need to solve here in the next PR if I use the `procMu` lock, as I want `p.Start` to be synchronous, yet it also wants the lock.

Reply:
I guess we can have a private method `p.start()` that assumes the lock is held, that can be called from here